scripts/interpreter.js

```diff
@@ -186,10 +186,17 @@
         for (let i = 0; i < robots.length; i++) {
             let robotFuncs = new RobotFunctions(intptr.robots[i], ball, i);
             let startFuncs = new DummyRobotFunctions(intptr.robots[i], ball, i);
-            let interpreter = new Interpreter(codes[i], intptr.generateInitFunc(robotFuncs));
-            let startInterpreter = new Interpreter(codes[i], intptr.generateInitFunc(startFuncs));
-            intptr.interpreters.push(interpreter);
-            intptr.startInterpreters.push(startInterpreter);
+            try {
+                let interpreter = new Interpreter(codes[i], intptr.generateInitFunc(robotFuncs));
+                let startInterpreter = new Interpreter(codes[i], intptr.generateInitFunc(startFuncs));
+                intptr.interpreters.push(interpreter);
+                intptr.startInterpreters.push(startInterpreter);
+            } catch (e) {
+                errorHandler.addError(`Error (Robot ${i + 1}): ${e.toString()}`, 'danger');
+                // push placeholders anyway
+                intptr.interpreters.push(null);
```

Comment:
Does this mean that execution will continue onto the next line anyway regardless of the current line error?

Reply:
I push `null` for interpreters that don't exist to make sure the robot numbers are still correct.

If I don't push null, then my robot number might be off by one (because the array would be like, `[interpreter]` rather than `[null, interpreter]`.